### ISSUE TITLE
TypeError when mouse component "clickable stimuli" only contains one object

### ISSUE THREAD

**Issue by bcmross** [psychopy/psychopy#3754]
For PsychoPy version 2021.1.4  (a different error for this was recently resolved in 2021.1.3 #3651 #3656)

**Situation:** I have a rectangle polygon object named `click_here` to act as a button to end the routine. I also have a mouse component and list `click_here` as the only object in "clickable stimuli" and end routine set to valid click.

**Error:** 
```
Traceback (most recent call last):
  File "---", line ---, in <module>
    for obj in click_here:
TypeError: 'Rect' object is not iterable
```

**Issue in code:** When multiple objects are listed in clickable stimuli, a list object is in place of `click_here`, like `[click_here, click_here_2]`, and there is no error thrown because the list type is iterable. But when a single object is listed, it is given as the object itself, like `click_here`, but still needs to be given in the form of a list, like `[click_here]`.

**Temporary fix:** Add a comma after your object name in clickable stimuli, like `click_here,`. That seems to force it as a list object.

**Comment by TEParsons** [comment]
The dilemma here is: What do we do when the object is a `list` variable? For example, if your earlier code said:
```
click_here = [myStim1, myStim2, myStim3]
```
and you had set the clickable stimuli to be `click_here`, you wouldn't want it encasing in `[]` as you would then have `[[myStim1, myStim2, myStim3]]`. However, in the case you describe, you would want it encasing in `[]` as it needs to be iterable later on.

Currently, the code is set up so that you can do exactly as you did in your temporary fix: Either add `[]` or a comma to indicate that the variable name is an item in a list, not the name of a list. The alternative would be to have this presumed to be the case, in which case people wanting to use a list would be stuck (as was the case in #3651).

I'm confident that there's a way to handle this which suits both cases - even if it's just a case of better communicating (perhaps in the tooltip) that there needs to be a `,` to indicate a single-item list.

**Comment by peircej** [comment]
In 2021.2 (imminent) we check at runtime if whatever has been provided is iterable and convert it to a list then. This has been implemented here: bac1692bb3aa638c604df05eb28edb95b57cd18e 

Ideally, we should do this slightly more neatly within the Mouse class. I'm thinking we should add a method to that called `getUnderlyingObjects(listOfObjects)`. Then we can test for a click and if we find one then underlyingObjetcs become clickedObjects. The method `getUnderlyingObjects` will handle the conversion to iterable if not already so that script code looks neater

**Comment by peircej** [comment]
Look out for that method in 2022.1. For now though I think we can close this as is it at least fixed in the Builder-generated scripts (thanks @RebeccaHirst )
